FreeRTOS/FreeRTOS-Partner-Supported-Demos
# Adding demos for FreeRTOS port for c28x based microcontrollers

Adding demos for FreeRTOS port for c28x based microcontrollers

Description
-----------
This pull request consists of demo projects for TMS320F2838xD (C2000 compiler 20.2.1.LTS) for the TMDSCNCD28388D controlcard board.

Test Steps
-----------
To run the demos on TMDSCNCD28388D controlcard, the following steps are required:
 - Install Code Compser Studio(CCS): https://www.ti.com/tool/CCSTUDIO
 - Install latest C2000Ware package: https://www.ti.com/tool/C2000WARE
 - Update the C2000WARE_ROOT and FreeRTOS_ROOT path variable in the example projectspec files available at following location in the repo: FreeRTOS-Partner-Supported-Demos\C2000_F2838x_C28x_CCS\CCS
 - Import the example project file in CCS
 - Build and debug the project

Standard Test Demo Results
-----------
![image](https://user-images.githubusercontent.com/93313481/162381617-0fb0c63f-9778-4c16-99b0-677737551911.png)

Known Issues
-----------
Support for "fpu64" is not added yet to the port. The examples should specify "fpu32" as option for floating point.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
